scheduler/src/cook/mesos/rebalancer.clj

```diff
@@ -299,12 +343,14 @@
 
 (defn rebalance
   "Takes a db, a list of pending job entities, a map of spare resources and params.
-   Returns a list of pending job entities to run and a list of task entities to preempt"
-  [db pending-job-ents host->spare-resources {:keys [max-preemption] :as params}]
+   Returns a list of pending job entities to run and a list of task entities to preempt
+
+   category is :normal or :gpu, depending on which type of job we're working with"
+  [db pending-job-ents host->spare-resources {:keys [max-preemption category] :as params}]
```

Comment:
I can't seem to find where you are calling rebalance with different categories (except in tests). What am I missing?